docs/dataservice-read-catalog-example.md

```diff
@@ -44,241 +51,329 @@ After building and running the example, you see the following:
-* Specify the path to Android NDK's toolchain file set via the `CMAKE_TOOLCHAIN_FILE` variable.
-* In case you want to build SDK for a specific Android platform, use `-DANDROID_PLATFORM CMake` flag, and `-DANDROID_ABI`, if you want to build for specific Android architecture. For more details, see [NDK-specific CMake variables](https://developer.android.com/ndk/guides/cmake#variables).
+### <a name="build-sdk-android"></a>Build the HERE OLP SDK for C++
 
-```bash
-mkdir build && cd build
-cmake .. -DCMAKE_TOOLCHAIN_FILE=$NDK_ROOT/build/cmake/android.toolchain.cmake -DANDROID_ABI=arm64-v8a -DOLP_SDK_BUILD_EXAMPLES=ON -DOLP_SDK_ENABLE_TESTING=OFF
-```
+To build the HERE OLP SDK for C++ on Android:
 
-The `CMake` command generates a `Gradle` project in the `build/examples/dataservice-read/android` folder. Before using the project, install the HERE OLP SDK for C++ libraries into the sysroot directory:
+1. Set `OLP_SDK_BUILD_EXAMPLES` to `ON`.
+2. (Optional) To disable tests, set `OLP_SDK_ENABLE_TESTING` to `OFF`.
+3. Specify the path to the Android NDK toolchain file using the `CMAKE_TOOLCHAIN_FILE` variable.
+4. If you want to build the SDK for a specific Android platform, use the `-DANDROID_PLATFORM CMake` flag, and if you want to build the SDK for a specific Android architecture, use the `-DANDROID_ABI` flag.
+   For more details, see [NDK-specific CMake variables](https://developer.android.com/ndk/guides/cmake#variables).
 
-```bash
-# Execute as sudo if necessary
-(sudo) make install
-```
+   ```bash
+   mkdir build && cd build
+   cmake .. -DCMAKE_TOOLCHAIN_FILE=$NDK_ROOT/build/cmake/android.toolchain.cmake -DANDROID_ABI=arm64-v8a -DOLP_SDK_BUILD_EXAMPLES=ON -DOLP_SDK_ENABLE_TESTING=OFF
+   ```
 
-### Assemble APK and run it on Android device
+   The `CMake` command generates a `Gradle` project in the `build/examples/dataservice-read/android` folder.
 
-After you complete the `Gradle` project configuration, you can use `gradlew` executable to build and install the APK on your Android device:
+5. Install the HERE OLP SDK for C++ libraries into the **sysroot** directory.
 
-```bash
-cd examples/dataservice-read/android
-./gradlew assembleDebug
-./gradlew installDebug
-```
+   ```bash
+   # If necessary, execute as sudo.
+   (sudo) make install
+   ```
+
+### <a name="build-and-run-android"></a>Build and Run the APK
+
+To build and run the APK:
+
+1. In the Android Studio IDE, open the `build/examples/dataservice-read/android/build.gradle` script.
+2. Provide your application access key ID and access key secret.
+3. Install and run the `dataservice_read_example` APK.
+
+The main screen displays the following message: "Reading the partition data from the specified catalog completed successfully."
 
-Alternatively, you can use Android Studio IDE by opening the `build/examples/dataservice-read/android/build.gradle` script.
+If you encounter an error, in your device LogCat, find the error message.
```

Comment:
```suggestion
If you encounter an error, find the error message in your device Logcat.
```
The error doesn't happen in the Logcat, but in the application itself. Logcat just helps to debug this error message. As an alternative, we can get rid of this sentence.

Also, Locat, I guess, is not written in camel-case: https://developer.android.com/studio/command-line/logcat

Reply:
In this sentence, "in your device LogCat" indicates the place where the user should perform the action "find the error message". It doesn't denote the place where the error occurs as there is a comma in front of it. I would leave the sentence structure as is because the place of the action should go before the action (in tech writing).

Reply:
Deleted the sentence